### DISCUSSION TITLE
Not able to export data to desktop viewer

### DISCUSSION THREAD

**Discussion by guadalupe-duolingo** [CtrlSpice/otel-desktop-viewer#130] [Q&A]
Hi @CtrlSpice , I have the following in my config file to export telemetry to port 4317 and I started the desktop viewer with the `otel-desktop-viewer` command. 
`otel.metrics.exporter=none
otel.traces.exporter=logging,otlp
otel.exporter.otlp.traces.endpoint=host.docker.internal:4317
otel.propagators=tracecontext,baggage,jaeger
otel.traces.sampler=parentbased_traceidratio
otel.traces.sampler.arg=1
otel.javaagent.debug=true`
My application is running in a docker container which is why I have the endpoint as `host.docker.internal:4317` (when I tried using localhost:4317 I got a connection refused error).
However, when I run the application I don't see any traces appear on the `localhost:8000`.

Can anyone give me guidance as to how to make this work? Thanks.

**Comment by svrnm** [comment]
See https://github.com/CtrlSpice/otel-desktop-viewer/pull/129, this fixed your issue, since this is not yet part of a release you need to build otel-desktop-viewer from source:

```
FROM golang as BUILDER
RUN apt-get install -y ca-certificates git
RUN git clone https://github.com/CtrlSpice/otel-desktop-viewer.git /otel-desktop-viewer
RUN cd /otel-desktop-viewer ; go build

FROM debian:11-slim
COPY --from=BUILDER /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
COPY --from=BUILDER /otel-desktop-viewer/otel-desktop-viewer /bin/otel-desktop-viewer

ENTRYPOINT ["/bin/otel-desktop-viewer"]
EXPOSE 4317 55678 55679 8000
```

**Reply by guadalupe-duolingo** [reply]
Is there a way to just expose the port in docker so that it works without having to clone the entire repo in the build?

**Reply by svrnm** [reply]
please see the description of #129, without that patch otel-desktop-viewer exposes the port ONLY to localhost which in the case of the container is the container itself. You probably can do some container-trickery & magic to fix that as well, but I leave that to you to find out how this works
